### PR TITLE
[Marketplace Contribution] DetectAndDecodeBarcode Pack

### PR DESCRIPTION
**Pull request by limbernie**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
depends: pyzbar, pillow and zbar library (libzbar0)

## Description
Pack provides a script to detect and decode 2D barcodes (including QR codes) in a file using [pyzbar](https://pypi.org/project/pyzbar/) library.

## Use Case
Ideal for use in phishing playbooks where an organization wants to decode not just QR codes but other 2D barcodes as well in a file, and is not comfortable with uploading images to an external service for decoding. As noted in this Cofense [article](https://cofense.com/blog/malicious-actors-utilizing-qr-codes-to-deploy-phishing-pages-to-mobile-devices/), threat actors are increasingly using QR codes in phishing emails to bypass traditional file and text detection.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -26,8 +26,14 @@ RUN VERSION="lts" bash /features/src/node/install.sh
...
+# install zbar library
+RUN apt-get update && apt-get install -y \
+    libzbar0
+
 # install poetry, configure certificate for git and tools for oh my zsh
 RUN python -m pip install --user pipx \
+    && python -m pip install pyzbar \
+    && python -m pip install pillow \
     && python -m pipx ensurepath \
     && pipx install poetry \
```
Comment (ostolero) [inline .devcontainer/Dockerfile:38]: Revert these changes and create a docker pull request.

**Review by ostolero — COMMENTED** [review]
Hi @limbernie , thank you for this contribution looks like a useful feature!
The script looks good, but as it require packages that are not currently available in any of our dockers, we will need to create another contribution to our docker repo.
From what I saw the best docker to use will be `processing-image-file` which already has the `pillow` package. Please add the other two packages and open a pr as guided in our Readme https://github.com/demisto/dockerfiles
Once we have a docker pr we can go ahead and demo the pack.
Let me know if you need any help with that I'm available here and on Slack.

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -26,8 +26,14 @@ RUN VERSION="lts" bash /features/src/node/install.sh
...
     && git clone https://github.com/zsh-users/zsh-syntax-highlighting.git $HOME/.zsh/zsh-syntax-highlighting \
```
[the remark this answers is not in the document]
Comment (limbernie) [reply .devcontainer/Dockerfile:39]: Sorry, was busy with work. Will revert the changes and create a docker PR. Thanks for the heads up!

**Comment by ostolero** [comment]
Great @limbernie, please tag me in the docker pr when it's available.

**Comment by ostolero** [comment]
Hi @limbernie , looks like our docker pr is good to go! now if you can add the new docker in the yml - `devdemisto/processing-image-file:1.0.0.75749`
Let's make sure all tests pass and then we can go ahead and schedule our demo.

**Comment by ostolero** [comment]
Hi @limbernie , thank you for your patience, we are taking longer to response we the current situation in Israel and hope to get this merged as soon as possible.
I see the validation fail due to not finiding the dev docker, I've merged the docker pr and I think this should fix the issue. could you change the docker to `demisto/processing-image-file:1.0.0.76977` ?

**Comment by limbernie** [comment]
> Hi @limbernie , thank you for your patience, we are taking longer to response we the current situation in Israel and hope to get this merged as soon as possible.
I see the validation fail due to not finiding the dev docker, I've merged the docker pr and I think this should fix the issue. could you change the docker to `demisto/processing-image-file:1.0.0.76977` ? 

Hi @ostolero,  noted with thanks. No worries and stay safe.

**Comment by ostolero** [comment]
Hi @limbernie, thank you for your patience, Looks like the docker works as expected.
I've reached out through email to schedule our demo.
There are still some fixes to be done to make sure validation passes-
1. Ut - we require a unit test coverage of 70% and your currently is 64%, add another UT for the code that is not currently tested to increase coverage.
2. There are a couple of secrets detected in your code, remove them or place them in the .secrets-ignore file

**Comment by ostolero** [comment]
Hi @limbernie , any updates on this pr?

**Comment by omerKarkKatz** [comment]
@limbernie Hey thanks a lot for the changes towards merge.
Please also look at the `run validation` task as part of the GitHub actions build. 
You failed the secrets step if these are not secrets please insert them into the .secret_ignore file. https://xsoar.pan.dev/docs/packs/packs-format#secrets-ignore:~:text=.secrets%2Dignore-,%23,-This%20file%20will

After that, we will be ready for a demo. 
For the Demo it is possible to send a video explaining the use case and executing the commands added or schedule a meeting demo please let me know what works better for you. 
If you prefer a meeting please let me know what time slots are available for you next week, via Slack DFIR or email. 
And if you prefer a video feel free to pass it over by email or Slack as well. 

For more info on the demo process please refer to:
https://xsoar.pan.dev/docs/contributing/demo-prep

BTW are you planning on further work on this pack? 
If not i will most probably add it in the inner PR to the pack called CommunityCommonScripts.

Thanks a lot please feel free to reach out if you have any further questions or if you need further help.
@omerKarkKatz

**Comment by omerKarkKatz** [comment]
Hi @limbernie, we haven’t heard from you in a while.
Do you need any help with the pull request?

Please don't hesitate to reach out to me here or on Slack.
Thanks again for contributing to our repo, I hope to hear from you soon.

**Comment by omerKarkKatz** [comment]
Hi @limbernie,
Thank you for your contribution. 
I have not received a response from you in the last two weeks, so I will close your PR now.
Please feel free to re-open it when you are available to continue. 
Thanks again.
